crates/wasmtime/src/values.rs

```diff
@@ -94,7 +101,13 @@ impl Val {
                     .insert_with_gc(x.inner, store.stack_map_registry());
                 ptr::write(p as *mut *mut u8, externref_ptr)
             }
-            _ => unimplemented!("Val::write_value_to"),
+            Val::FuncRef(None) => ptr::write(p, 0),
```

Comment:
I think this should write the store "null funcref" singleton instead of a null pointer, right?